File: app/src/main/java/app/islammedia/halaqatalquran/Fragments/StudentsViewFragment.java
package app.islammedia.halaqatalquran.Fragments;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.room.Room;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

import app.islammedia.halaqatalquran.Adapters.StudentsAdapter;
import app.islammedia.halaqatalquran.R;
import app.islammedia.halaqatalquran.Database.MainDataBase;
import app.islammedia.halaqatalquran.Database.Student;


/**
 * A simple {@link Fragment} subclass.
 */
public class StudentsViewFragment extends Fragment {

    public StudentsViewFragment() {
        // Required empty public constructor
    }

    RecyclerView stRv;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_students_view, container, false);
        stRv = v.findViewById(R.id.studentsViewList);
        stRv.setHasFixedSize(true);
        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(v.getContext());
        stRv.setLayoutManager(layoutManager);
        new Thread(getStList).start();
        return v;
    }

    Runnable getStList = () -> {
        if (getActivity()==null) return;
        MainDataBase db = Room.databaseBuilder(getActivity(),MainDataBase.class,"MainDataBase").build();
        List<Student> sts = db.myDAO().getAllStudents();
        StudentsAdapter stAdapter = new StudentsAdapter(sts,getActivity());
        getActivity().runOnUiThread(()-> {
            stRv.addItemDecoration(new DividerItemDecoration(getActivity(), DividerItemDecoration.VERTICAL));
            stRv.setAdapter(stAdapter);
        });
    };
}

File: app/src/main/java/app/islammedia/halaqatalquran/fragments/StudentsViewFragment.kt
package app.islammedia.halaqatalquran.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import app.islammedia.halaqatalquran.adapters.StudentsAdapter
import app.islammedia.halaqatalquran.R
import app.islammedia.halaqatalquran.database.MainDataBaseManager

class StudentsViewFragment : Fragment() {
    private lateinit var allStudentsView: RecyclerView
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(R.layout.fragment_students_view, container, false)
        allStudentsView = v.findViewById(R.id.studentsViewList)
        allStudentsView.setHasFixedSize(true)
        val layoutManager: RecyclerView.LayoutManager = LinearLayoutManager(v.context)
        allStudentsView.layoutManager = layoutManager
        Thread(getStudentsList).start()
        return v
    }

    private var getStudentsList: Runnable = Runnable {
        if (activity == null) return@Runnable

        val db = MainDataBaseManager.getMainDataBaseInstance(requireActivity())


        val students = db.myDAO().getAllStudents()
        val studentsAdapter = StudentsAdapter(students, requireActivity())
        requireActivity().runOnUiThread {
            allStudentsView.addItemDecoration(DividerItemDecoration(activity, DividerItemDecoration.VERTICAL))
            allStudentsView.adapter = studentsAdapter
        }
    }
}